### PR TITLE
adding the --end-with and --only-package options

### PR DESCRIPTION
**Pull request by wjwwood**
This pr adds two new options to the `ament build` verb, `--end-with` and `--only-package`.

Examples:

```
% ament build --start-with foo --end-with bar
% ament build --end-with bar
% ament build --only-package foo
# Which is equivalent to:
% ament build --start-with foo --end-with foo
```

### CODE REVIEW

**Comment by wjwwood** [comment]
@dirk-thomas @esteve @tfoote for review.

File: ament_tools/verbs/build/cli.py [hunk 1]
```diff
@@ -100,6 +100,15 @@ def prepare_arguments(parser, args):
         '--start-with',
         help='Start with a particular package',
     )
+    parser.add_argument(
+        '--end-with',
+        help='Start with a particular package',
```
Comment (esteve) [inline ament_tools/verbs/build/cli.py]: s/Start/End/

**Comment by esteve** [comment]
+1

File: ament_tools/verbs/build/cli.py (hunk 1 above)
Comment (wjwwood) [inline ament_tools/verbs/build/cli.py]: Oops, good catch.

File: ament_tools/verbs/build/cli.py [hunk 2]
```diff
@@ -100,6 +100,15 @@ def prepare_arguments(parser, args):
         '--start-with',
         help='Start with a particular package',
     )
+    parser.add_argument(
+        '--end-with',
+        help='End with a particular package',
+    )
+    parser.add_argument(
+        '--only-package',
+        '--only',
```
Comment (dirk-thomas) [inline ament_tools/verbs/build/cli.py]: NItpick: Flip the order of the two options. Similar to `-C` / `--directory`.

File: ament_tools/verbs/build/cli.py [hunk 3]
```diff
@@ -141,19 +150,53 @@ def main(opts, per_package_main=build_pkg_main):
 def print_topological_order(opts, packages):
     package_names = [p.name for _, p, _ in packages]
 
+    if opts.only_package and (opts.start_with or opts.end_with):
+        # The argprase mutually exclusive mechanism is broken for subparsers
+        # See: http://bugs.python.org/issue10680
+        # So we'll check it manually here
+        sys.exit("The --start-with and --end-with options cannot be used with "
+                 "the --only-package option.")
+
```
Comment (dirk-thomas) [inline ament_tools/verbs/build/cli.py]: I would move this check inside the condition from line 168.

File: ament_tools/verbs/build/cli.py [hunk 4]
```diff
@@ -190,6 +233,9 @@ def iterate_packages(opts, packages, per_package_callback):
         rc = per_package_callback(opts)
         if rc:
             return rc
+        if package.name == opts.end_with:
+            print("Stopping with package '{0}'".format(package.name))
```
Comment (dirk-thomas) [inline ament_tools/verbs/build/cli.py]: I would phrase this:

    Stopped after package ...

File: ament_tools/verbs/build/cli.py (hunk 2 above)
Comment (wjwwood) [inline ament_tools/verbs/build/cli.py]: It only works with single `-` options, switching the order would change the destination name. I'll leave as is (we discussed it off-line).

**Comment by wjwwood** [comment]
Addressed.

**Comment by dirk-thomas** [comment]
+1

**Comment by wjwwood** [comment]
I'm going to merge, @tfoote let me know if you have retroactive comments.

**Comment by tfoote** [comment]
lgtm

**Pull request merged**